### PR TITLE
Rename mcmc functions and GetGlobalMode

### PR DESCRIPTION
**Pull request by fredRos**
This relates to #37 and #71 but still needs some work. Don't merge yet!

### CODE REVIEW

**Comment by oschulz** [comment]
Cleared myself as assignee  until it's ready for merge, to avoid merging by accident.

**Comment by fredRos** [comment]
This one is a major backward-incompatible change. `make distcheck` passes.

**Comment by oschulz** [comment]
Is this ready to be merged, from your side?

File: BAT/BCIntegrate.h [hunk 1]
```diff
@@ -372,7 +372,7 @@ class BCIntegrate : public BCEngineMCMC
 
     /**
      * @return vector of parameter and observable values at global mode. */
-    virtual const std::vector<double>& GetGlobalMode() const;
+    virtual const std::vector<double>& GetBestFitParameters() const;
```
Comment (oschulz) [inline BAT/BCIntegrate.h]: Nice!

**Comment by oschulz** [comment]
I like this a lot - the whole naming inside BCEngineMCMC is a lot saner now, and there was no reason to prefix constant and functions names within the class, anyhow.

**Comment by fredRos** [comment]
I'm glad you like it. Yes it's ready for merging. I sat together with @greenwald for two hours and discussed. This one will be the last major change to the code. There are a couple of small things left, the rest is documentation. I'm starting to see version 1.0

File: BAT/BCIntegrate.h (hunk 1 above)
Comment (fredRos) [inline BAT/BCIntegrate.h]: This is just reverting to the old name in 0.9.4 :smile:

**Pull request merged**